client/my-sites/plans-v2/product-card/index.tsx

```diff
@@ -148,6 +153,13 @@ const ProductCardWrapper = ( {
 	const isUpgradeableToYearly =
 		isOwned && selectedTerm === TERM_ANNUALLY && item.term === TERM_MONTHLY;
 
+	// Don't hide price if siteId is not defined, since it most likely won't be shown
+	// in other parts of the card (e.g. Jetpack Search)
+	// Jetpack CRM is an external product and we don't have access to its price, therefore,
+	// we hide it.
+	const hidePrice =
+		EXTERNAL_PRODUCTS_LIST.includes( item.productSlug ) || ( !! siteId && item.hidePrice );
```

Comment:
I'm not very pleased about this. I think we might have to refactor it, but no sure if this is the right PR to do that.